app/controllers/admin/suppliers_controller.rb

```diff
@@ -1,9 +1,15 @@
 class Admin::SuppliersController < AdminController
   def index
-    @suppliers = Supplier.page(params[:page])
+    @suppliers = Supplier.search(params[:search]).page(params[:page])
   end
 
   def show
     @supplier = Supplier.find(params[:id])
   end
+
+  private
+
+  def supplier_params
```

Comment:
These are only required when mass-assigning attributes, such as when creating a new record from a hash of parameters. We would need it if we were to create a new supplier. As we're not doing that yet, I suggest we don't add the code until we need it.

Reply:
Cool. This was one of those things I've seen but not fully understood.